### PR TITLE
[Refactor] Load use new Analyzer

### PR DESCRIPTION
**Pull request by kangkaisen**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/LoadScanNode.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/load/Load.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|10|10|100.00%|

**Comment by kangkaisen** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/LoadScanNode.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/load/Load.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|10|10|100.00%|

**Comment by Youngwb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 63 / 202 (31.19%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/ScalarOperatorToExpr.java|23|159|14.47%|
|:large_blue_circle:|com/starrocks/load/Load.java|6|9|66.67%|
|:large_blue_circle:|com/starrocks/planner/LoadScanNode.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/SlotRef.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorRewriter.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/SlotDescriptor.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/SqlToScalarOperatorTranslator.java|9|9|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|10|10|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 22 / 25 (88.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/Load.java|11|14|78.57%|
|:large_blue_circle:|com/starrocks/planner/LoadScanNode.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/analysis/SlotRef.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorRewriter.java|1|1|100.00%|

**Review by HangyuanLiu — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/load/Load.java [hunk 1]
```diff
@@ -610,12 +621,24 @@ private static void analyzeMappingExprs(Table tbl, Analyzer analyzer, Map<String
...
+        ExpressionAnalyzer.analyzeExpressionIgnoreSlot(expr, ConnectContext.get());
+        // Translating expr to scalar in order to do some rewrites
+        ScalarOperator scalarOperator = SqlToScalarOperatorTranslator.translate(expr);
+        ScalarOperatorRewriter scalarRewriter = new ScalarOperatorRewriter();
+        // Add cast and constant fold
+        scalarOperator = scalarRewriter.rewrite(scalarOperator, ScalarOperatorRewriter.LOAD_REWRITE_RULES);
+        requireNonNull(scalarOperator, "rewrite expression is null");
+        return ScalarOperatorToExpr.buildExprIgnoreSlot(scalarOperator,
+                new ScalarOperatorToExpr.FormatterContext(Maps.newHashMap()));
+
```
[the remark this answers is not in the document]
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/load/Load.java:640]: I will fix it in next PR

**Pull request merged**